src/metax_api/models/catalog_record.py

```diff
@@ -1130,18 +1139,19 @@ def _post_create_operations(self):
         if other_record:
             self._create_or_update_alternate_record_set(other_record)
 
-        if get_identifier_type(self.preferred_identifier) == IdentifierType.DOI:
-            self._validate_cr_against_datacite_schema()
-            self.add_post_request_callable(DataciteDOIUpdate(self, self.research_dataset['preferred_identifier'],
-                                                             'create'))
+        if not self.state_is_draft(self.request):
+            if get_identifier_type(self.preferred_identifier) == IdentifierType.DOI:
+                self._validate_cr_against_datacite_schema()
+                self.add_post_request_callable(DataciteDOIUpdate(self, self.research_dataset['preferred_identifier'],
+                                                                'create'))
 
-        if self._dataset_has_rems_managed_access() and settings.REMS['ENABLED']:
-            self._validate_for_rems()
-            user_info = self._get_user_info_for_rems()
-            self._access_granter = user_info
-            self.add_post_request_callable(REMSUpdate(self, 'create', user_info=user_info))
+            if self._dataset_has_rems_managed_access() and settings.REMS['ENABLED']:
+                self._validate_for_rems()
+                user_info = self._get_user_info_for_rems()
+                self._access_granter = user_info
+                self.add_post_request_callable(REMSUpdate(self, 'create', user_info=user_info))
 
-        self.add_post_request_callable(RabbitMQPublishRecord(self, 'create'))
+            self.add_post_request_callable(RabbitMQPublishRecord(self, 'create'))
```

Comment:
While all of this works, I recommend slightly re-organizing this part to be more obvious with something like:

```
        if self.state_is_draft(self.request):
            # do nothing
            pass
        else:

            # publishing directly

            if get_identifier_type(self.preferred_identifier) == IdentifierType.DOI:
            ...
```
Also, the block from line 1122 ``if self.catalog_versions_datasets(): ...`` should be moved inside this block too. The stuff from line 1138 on the other hand is fine where it is, since there it should not be possible that there are "alternate records" at this point.

Reply:
Ok, moved & re-organized